lib/Parse/ParseDecl.cpp

```diff
@@ -2031,6 +2031,9 @@ bool Parser::parseTypeAttribute(TypeAttributes &Attributes, bool justChecking) {
   SourceRange autoclosureEscapingParenRange;
   StringRef conventionName;
   StringRef witnessMethodProtocol;
+  // SWIFT_ENABLE_TENSORFLOW
+  StringRef differentiationMode;
```

Comment:
Is there some reason `differentiationMode` is stored as a string and not converted to an `AutoDiffMode`?

Reply:
Good catch, I shouldn't call this `differentiationMode`, but `differentiabilityName` instead. 

`AutoDiffMode` is the real "mode" here, and it should only have `Forward` and `Reverse`. Differentiability variants like `Linear` and `Constant` have nothing to do with the mode of differentiation.